src/services/global-state.ts

```diff
@@ -0,0 +1,13 @@
+import type { Firestore } from '@google-cloud/firestore';
+import type { CronJob } from 'cron';
+
+interface GlobalState {
+  activityCronJob?: CronJob;
+  firestore?: Firestore;
+  randomVoiceChannelNameCronMap: Map<string, CronJob>;
+}
+
+// TODO find a better way to do this
+export const GLOBAL_STATE: GlobalState = {
```

Comment:
I am probably going to create hook style reducer functions for these too, but not 100% sure.